app/app.go

```diff
@@ -142,8 +142,8 @@ func (a *App) initJobs() {
 		a.srv.Jobs.Cloud = jobsCloudInterface(a.srv)
 	}
 
-	a.srv.Jobs.Workers = a.srv.Jobs.InitWorkers()
-	a.srv.Jobs.Schedulers = a.srv.Jobs.InitSchedulers()
+	a.srv.Jobs.InitWorkers()
+	a.srv.Jobs.InitSchedulers()
```

Comment:
0/5, but I'd want to see a better API where the concept of schedulers is integrated into the job server. Right now, both are part of the same package but schedulers is inside the jobs and handled in a weird way.

I'd want to see the entire initial state of whether to start jobserver, or start jobserver and schedulers both in a `New` function, which returns an opaque object. And then starting it starts everything that's needed. And the caller does not need to bother about when to init, when to start, and whether to start workers or schedulers.

Completely out of scope for this PR though. I am wondering if you have any thoughts on this. Going through the PR, I found it tremendously difficult to create a mental model of the entire package.

Reply:
Well, this is a bit more than a patch but not a complete refactor either which is still due. I agree the architecture of the package being way more complex than needed with too many exported methods. I imagine all these things can be safely addressed in [MM-26769](https://mattermost.atlassian.net/browse/MM-26769).

Reply:
Agree.